### PR TITLE
update parallelproj to 1.9.0 (and require it), as well as CIL, SIRF, STIR

### PR DESCRIPTION
**Pull request by KrisThielemans**
See https://github.com/gschramm/parallelproj/issues/69

### CODE REVIEW

**Comment by KrisThielemans** [comment]
Failures due to issues unrelated to this PR:
https://github.com/SyneRBI/SIRF/issues/1267
https://github.com/TomographicImaging/CIL/issues/1837
https://github.com/TomographicImaging/CIL/issues/1838

**Comment by KrisThielemans** [comment]
We need to update SIRF and CIL for numpy 2.0 issues. Can just as well do STIR.

**Review by casperdcl — APPROVED** [review]

**Comment by KrisThielemans** [comment]
2 problems resolved, but `CIL_FRAMEWORK_TESTS_6 ` still fails due to https://github.com/TomographicImaging/CIL/issues/1838

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # ChangeLog
 
+## v3.8.0
+- Updated Versions:
+  - SIRF: 81f5c0a7878ed60577ee526abf8d0cd2b64e334f (20 June 2024)
+  - STIR: 12bfa873522653936a4818e9bad9b9da41f11706 (20 June 2024)
+  - parallelproj: 1.9.1 (contains a fix for truncation of the TOF kernel)
+  - CIL: 8d111a37cec72756d4b8d80f526ff22333019037 (20 June 2024, numpy 2.0 fix)
```
Comment (casperdcl) [inline CHANGES.md:8]: should be updated after https://github.com/TomographicImaging/CIL/pull/1844

File: version_config.cmake [hunk 2]
```diff
@@ -161,7 +161,7 @@ set(DEFAULT_JSON_TAG v3.11.3)
 # CCPi CIL
 # minimum supported version of CIL supported is > 22.1.0 or from commit a6062410028c9872c5b355be40b96ed1497fed2a
 set(DEFAULT_CIL_URL https://github.com/TomographicImaging/CIL)
-set(DEFAULT_CIL_TAG ccf17f393ba911d13b74f2327779dde030098fe6) # 28 May 2024
+set(DEFAULT_CIL_TAG 8d111a37cec72756d4b8d80f526ff22333019037) # 20 June 2024 (numpy 2.0 fix)
```
Comment (casperdcl) [inline version_config.cmake:164]: should be updated after https://github.com/TomographicImaging/CIL/pull/1844

File: CHANGES.md (hunk 1 above)
Comment (casperdcl) [reply CHANGES.md:8]:
```suggestion
  - CIL: 8449f663bb585c3f5744a7aff78937b6f12d7e20 (21 June 2024)
```

File: version_config.cmake (hunk 2 above)
Comment (casperdcl) [reply version_config.cmake:164]:
```suggestion
set(DEFAULT_CIL_TAG 8449f663bb585c3f5744a7aff78937b6f12d7e20) # 21 June 2024
```

**Comment by KrisThielemans** [comment]
Some failures due to Zenodo time-out. Docker jobs
```
ERROR: test_SIRF (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_SIRF
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/opt/conda/lib/python3.10/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/jovyan/cil_sirf_test/test_SIRF.py", line 29, in <module>
    from cil.optimisation.utilities import AdaptiveSensitivity
ImportError: cannot import name 'AdaptiveSensitivity' from 'cil.optimisation.utilities' (/opt/conda/lib/python3.10/site-packages/cil/optimisation/utilities/__init__.py)
```
not sure if this is the most recent though.

**Comment by KrisThielemans** [comment]
https://github.com/TomographicImaging/CIL/issues/1846

**Comment by KrisThielemans** [comment]
I'll merge this anyway, as it fixes other problems.

**Pull request merged**